pull-request/automerge/action.yml

```diff
@@ -0,0 +1,41 @@
+---
+name: Automerge Pull Requests
+description: |
+  Automatically merge pull requests.
+  The action fetches all open pull requests with a specific label (default: `autorelease: pending`) and merges them.
```

Comment:
why not use the existing approach in https://github.com/camunda/infra-core/blob/4fa292ff8293c95ad50aab7838b7722ebd37f5e5/.github/workflows/automerge-master-pr.yml#L2?

Reply:
hmm... I really tried several things now, but I can't get it working with the proposed solution:
https://github.com/kellervater/c8sm/actions/runs/11463419684/job/31897218278

it just doesn't recognize the [release-please PR](https://github.com/kellervater/c8sm/pull/38) in my test repository. Already put some time into that. Not sure if I'm missing something super obvious here.
but tbh. I feel like I'm currently wasting a lot of time here.

Reply:
tried different labels because I thought the space of `autorelease: pending` label is the problem.
Tried a direct call instead of the wrapper because I thought there may be a contextual problem. nothing helped so far.

Reply:
@kellervater can you invoke the workflow in a debug mode?

Reply:
yes. but it just doesn't tell me WHY it's not picking up on the PR. I'm currently trying different parameter combinations.

here a debug log: https://github.com/kellervater/c8sm/actions/runs/11463714585/job/31898205486

Reply:
Also an almost 1:1 copy if infra-core's automerge workflow doesn't bring the expected result:
https://github.com/kellervater/c8sm/actions/runs/11463898128/job/31898836829

I suspect, that it doesn't work, when not triggered by a pull request event.

Reply:
it may also be, because the label gets introduced by another bot. There's an issue hinting at that (but with some different use-case though):
https://github.com/pascalgn/automerge-action/issues/131

Reply:
I think it's because the action behaves differently based on what trigger happened: https://github.com/pascalgn/automerge-action/blob/f981436038b98090c5ad6b89718eeee5baa9847f/lib/api.js#L90

The workflow dispatch one calls this handler, with the Octokat PR filter looking suspicious (where's the repo name?):  https://github.com/pascalgn/automerge-action/blob/f981436038b98090c5ad6b89718eeee5baa9847f/lib/api.js#L249

Reply:
I'm currently unable to fix that problem with the third party action in a reasonable time (maybe too late already).
I could try to debug deeper tomorrow, but tbh: There was a quite simple and working solution in the first place, which should do just fine.
@maxdanilov do you think it's worth investigating further? Then I'll continue to look for a solution on the third-party automerge action tomorrow.

Reply:
> The workflow dispatch one calls this handler, with the Octokat PR filter looking suspicious (where's the repo name?): https://github.com/pascalgn/automerge-action/blob/f981436038b98090c5ad6b89718eeee5baa9847f/lib/api.js#L249

hmm... good hint. will try to debug further

Reply:
Looking into the REST API [doc](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests) of GitHub, I'm 100% sure that line is responsible one. The `head` value it passes looks like `kellervater:main` and the filter returns nothing. To capture the [PR](https://github.com/kellervater/c8sm/pull/38), the `head` value should've been `kellervater:release-please--branches--main`. Looks like a bug in the code (and I don't see why the `head` filter is needed in this case at all).

Reply:
Before debugging, try just whether it works properly with the cronjob event - if that is the main one, this may be sufficient already.

Reply:
> Try with the cronjob trigger, looking at the code it should work.

did that: https://github.com/kellervater/c8sm/actions/runs/11464405607/job/31900517435
no change

will continue tomorrow with an in-depth debugging session then :-)

Reply:
or not. it seems like it recognized the last one as workflow_dispatch again. strange.

Reply:
worked https://github.com/kellervater/c8sm/actions/runs/11464676291 🎉  just needs `MERGE_REQUIRED_APPROVALS` unspecified or 0.

Reply:
you were right. This time  it at least found the PR: https://github.com/kellervater/c8sm/actions/runs/11464676291/job/31901404272

Maybe worth opening an issue at the vendor, so they can fix the workflow_dispatch behavior (or do it ourselves and hope for a merge). 

Will try to continue with this approach now.

Reply:
the third(rty) time is the charm ;-)
https://github.com/kellervater/c8sm/actions/runs/11464837600/job/31901942700

thx @maxdanilov for your advise. Will use this approach now :-)